src/main/scala/is/hail/utils/richUtils/RichDenseMatrixDouble.scala

```diff
@@ -14,22 +15,55 @@ object RichDenseMatrixDouble {
       Some(DenseMatrix.horzcat(ms: _*))
   }
   
+  // copies n doubles as bytes from data to dos
+  def writeDoubles(dos: DataOutputStream, data: Array[Double], n: Int) {
+    assert(n <= data.length)
+    var nLeft = n
+    val bufSize = if (nLeft < (1 << 14)) nLeft else 1 << 14 // up to 8k doubles
```

Comment:
math.min(nLeft, 1 << 14)

1 << 14 is 16K, FYI.  Might be clearer to write this as 8 * 1024, say (8192 is also fine in my book.)